src/System.Private.CoreLib/src/Internal/Runtime/Augments/EnvironmentAugments.Unix.cs

```diff
@@ -30,7 +30,89 @@ private static void SetEnvironmentVariableCore(string variable, string value)
             if ("".Length != 0)
                 throw new NotImplementedException(); // Need to return something better than an empty environment block.
 
-            return Array.Empty<KeyValuePair<string,string>>();
+            unsafe
+            {
+                // Get byte** of environment variables from native interop
+                var unsafeBlock = (byte**)Interop.Sys.EnvironGetSystemEnvironment();
+                if (unsafeBlock == (byte**)0)
```

Comment:
I think we should just return and assume empty environment if this returns null.

Reply:
Done via rework.